src/main/java/xyz/nucleoid/dungeons/dungeons/game/loot/weapon/DgBow.java

```diff
@@ -25,22 +26,42 @@ public DgBow(DgBowType type, DgLootGrade grade, DgBowMaterial material, double a
         this.attackDamage = attackDamage;
     }
 
+    public static void registerModels() {
+        for (DgBowType type : DgBowType.values()) {
+            for (DgBowMaterial material : DgBowMaterial.values()) {
+                DgModelRegistry.register(type.asVanillaItem(), material.id, type.id);
+            }
+        }
+    }
+
     // TODO(weapons): flavour text
     public static DgBow generate(Random random) {
-        DgBowType type = DgBowType.choose(random);
-        DgBowMaterial material = DgBowMaterial.choose(random);
-        DgLootGrade grade = DgLootGrade.chooseInRange(random, material.minGrade, material.maxGrade);
-        double attackDamage = (type.baseDamage + (random.nextDouble() / 2)) * material.damageModifier * grade.damageModifier;
+        return create(null, null, null, random);
+    }
 
+    public static DgBow create(DgBowType type, DgBowMaterial material, DgLootGrade grade, Random random) {
```

Comment:
IMO this should just use the constructor. Since attack damage doesn't matter for the model examples, just use `new DgBow(type, material, grade, 0.0)`. Same goes for the rest, and then keep `generate` as is, deleting `create`